pkgs/development/haskell-modules/configuration-common.nix

```
@@ -1898,4 +1898,19 @@ EOT
     })
   ];
 
+  bson = overrideCabal super.bson (drv: {
+    # We replace 'network' by 'network-bsd' because cabal2nix cannot correctly that pick up.
+    libraryHaskellDepends = with pkgs.haskellPackages; [
+      base
+      binary
+      bytestring
+      cryptohash-md5
+      data-binary-ieee754
+      mtl
+      network-bsd # network-bsd AND NOT network
+      text
+      time
+    ];
+  });
```

This should be much simpler and not require us to maintain a list of dependencies

```suggestion
  bson =  appendConfigureFlag (super.bson.override { network = self.network-bsd; }) "-f-_old_network";
```